### PR TITLE
Changing activeClass, added disabled state, moved to Mixin, ember-cli 2.4.1, and more!

### LINKED ISSUE #3: Changing active class
Hi, thanks for this small helper. It's great. Would it be possible for it to take the default activeClass?

We normally change `active` to `is-active` in our apps.

``` JavaScript
Ember.LinkView.reopen({
    activeClass: 'is-active'
});
```

### LINKED ISSUE #10: Proposed changes, and how to structure PR's
I have a few proposed changes to submit but don't know how to structure the PR's. The main point of contention is moving main functionality to a Mixin. But here are my proposed changes;
1. Move main functionality to a Mixin, that way this could be use in other addon components that might already provide a dropdown component
2. Enable changing of the active class name, to solve issue #3 
3. Add in a "disabled" state/class when ALL child links are also disabled
4. Enable changing of the link jQuery selector, in case `{{link-to}}` tagName is changed
5. Move the `childLinkViews` build code out of `didRender`, apparently there are questions on how hooks are ran when defined in a Mixin but overridden in the Component (more research needed)
6. Move away from the "active" property name, causes problems when used with other addon components (at least it did for me)

Here is a gist of my proposed changes for review. Again, I'll submit PR's after some discussion and best approach for PR structure.

https://gist.github.com/Panman8201/ddb4be922143d9681078

### PR DESCRIPTION
**Pull request by Panman82**
_Finally_ got around to creating the PR for issue #10. This encompasses everything mentioned there, with your concerns resolved (I'm hoping anyway). Here are the commit messages and details:

**Update to ember-cli 2.4.1**
Updated the ember-cli dep to 2.4.1 and ran `ember init` to update other deps and file changes.

**Remove ember-data**
Removed `ember-data` from the `devDependencies`, not needed for this addon..

**Remove un-needed .gitkeep files**
Really minor, just removing un-needed `.gitkeep` files since the addon and app folders contain stuff.

**Make 'childLinkViews' an Ember Array**
Make the `childLinkViews` property an Ember Array, instead of having the Computed Properties convert it (setup for future commits). This commit includes a few other related changes;
- Create the initial array in the `init()` hook
- Run `_super()` on the `didRender()` hook
- Change `var` to `let` for variable declarations

**Add ability to change the 'activeClass' name**
Previously, the active class name was hard coded to ‘active’. Now it will pull the child `{{link-to}}` `activeClass` name by default. Or, users can specify what it will be by defining the `activeClass` on the `{{active-link}}`. Also added tests for this and updated the README (which has been expanded to show more examples).

**Add support for the disabled state**
As a negative to active, add a disabled state that adds a class when ALL child links are also disabled. Similarly, the class name draws from the child `{{link-to}}` but can be overridden. Users can always opt-out by defining `disabledClass=false`. Also added tests for this and updated the README.

**Expose the child link selector**
While probably a rare case, expose the selector used to find child links. That way if folks are using `<button>`s instead, they could update the `linkSelector`. Also added tests for this and updated the README.

**Move functionality to mixin**
By moving the main functionality of this addon into a Mixin, Components from other addons can implement these features. For example, other UI libraries might provide a dropdown Component where this might be a helpful addition.

**Minor README tweaks**
Just a few tweaks to the README; broke-up long lines, extra spacing between headers, added the Development section (in new addons), adjusted the tests link, and removed the Building section (doesn’t make sense for this addon).

Impacts on existing issues and PRs:
Resolves #3 
Resolves #10 
Closes #8

### CODE REVIEW

**Comment by Panman82** [comment]
@alexspeller Just want to ping ya for feedback. Been a couple weeks now and just didn't want to lose sight of this PR. I'll jump on Slack for a few hours if you'd rather chat. Again, no hurry, just keeping this on my todo list.

Edit: Just realized you're probably offline for the day (with the time difference and all). I'll try to be online tomorrow morning at some point.

File: addon/components/active-link.js [hunk 1]
```diff
@@ -2,23 +2,28 @@ import Ember from 'ember';
...
   classNameBindings: ['active'],
 
   active: Ember.computed('childLinkViews.@each.active', function() {
-    return Ember.A(this.get('childLinkViews')).isAny('active');
+    return this.get('childLinkViews').isAny('active');
   }),
 
-  didRender: function() {
+  init() {
+    this._super( ...arguments );
```
Comment (alexspeller) [inline addon/components/active-link.js]: There is no need to call super here. `init` on Ember.Component is an empty function.

File: addon/components/active-link.js [hunk 2]
```diff
@@ -2,23 +2,28 @@ import Ember from 'ember';
...
+    this.set('childLinkViews', Ember.A());
+  },
+
+  didRender() {
+    this._super( ...arguments );
```
Comment (alexspeller) [inline addon/components/active-link.js]: There is no need to call super here. There is no super `didRender` hook to call on Ember.Component

File: addon/components/active-link.js [hunk 3]
```diff
@@ -24,6 +20,19 @@ export default Ember.Component.extend({
...
-  }
+  },
+
+  hasActiveLinks: Ember.computed('childLinkViews.@each.active', function(){
+    return this.get('childLinkViews').isAny('active');
+  }),
+
+  activeClass: Ember.computed('childLinkViews.@each.active', function(){
+    let activeLink = this.get('childLinkViews').findBy('active');
+    return (activeLink ? activeLink.get('active') : 'active');
```
Comment (alexspeller) [inline addon/components/active-link.js]: This is great, it's good that it takes it from the child by default or you can override it

**Comment by alexspeller** [comment]
Really sorry this took so long to merge, it's a great PR and I have just been incredibly ill. Thanks!

**Comment by alexspeller** [comment]
version `0.1.0` is pushed to npm now :)

**Pull request merged**